### PR TITLE
wip

### PR DESCRIPTION
**Pull request by vsavkin**

### CODE REVIEW

File: modules/angular2/src/core/di/binding.ts [hunk 1]
```diff
@@ -49,21 +49,21 @@ const _EMPTY_LIST = CONST_EXPR([]);
...
  *
- * expect(injector.get(String)).toEqual('Hello');
+ * expect(injector.get("message")).toEqual('Hello');
  * ```
  */
 @CONST()
 export class Binding {
   /**
-   * Token used when retrieving this binding. Usually the `Type`.
+   * Token used when retrieving this binding. Usually, it is a type `Type`.
```
Comment (jeffbcross) [inline modules/angular2/src/core/di/binding.ts]: `Type` should be `{@link Type}`

File: modules/angular2/src/core/di/binding.ts [hunk 2]
```diff
@@ -92,26 +91,30 @@ export class Binding {
    * expect(injectorAlias.get(Vehicle)).toBe(injectorAlias.get(Car));
    * expect(injectorAlias.get(Vehicle) instanceof Car).toBe(true);
    * ```
+   *
+   * See full working example: http://plnkr.co/edit/RSTG86qgmoxCyj9SWPwY?p=preview
```
Comment (jeffbcross) [inline modules/angular2/src/core/di/binding.ts]: Update to new format described [here](https://docs.google.com/document/d/13kQ49My6ThynHOrT3s1cWII5RhNHHGSbOXffbQmQlc8/edit#heading=h.qvj8ll9k9t56)

File: modules/angular2/src/core/di/binding.ts [hunk 3]
```diff
@@ -92,26 +91,30 @@ export class Binding {
...
    * var injector = Injector.resolveAndCreate([
-   *   new Binding(String, { toValue: 'Hello' })
+   *   new Binding("message", { toValue: 'Hello' })
    * ]);
    *
-   * expect(injector.get(String)).toEqual('Hello');
+   * expect(injector.get("message")).toEqual('Hello');
    * ```
+   *
+   * See full working example: http://plnkr.co/edit/UFVsMVQIDe7l4waWziES?p=preview
```
Comment (jeffbcross) [inline modules/angular2/src/core/di/binding.ts]: See above comment

File: modules/angular2/src/core/di/binding.ts [hunk 4]
```diff
@@ -201,12 +203,13 @@ export class Binding {
     this._multi = multi;
   }
 
+  //TODO: Provide a full working example after alpha38 is released.
   /**
-   * Used to create multiple bindings matching the same token.
+   * Creates multiple bindings matching the same token (a multi-binding).
```
Comment (jeffbcross) [inline modules/angular2/src/core/di/binding.ts]: Would be nice to add info about when this is useful, as well as how frequently one might want to use multi-bindings.

Comment (jeffbcross) [inline modules/angular2/src/core/di/binding.ts]: Also, should add `Multi-Binding` to the glossary https://docs.google.com/document/d/1hBjD4u9alL6t0110Xi16t3grQOGFPBNprJhV8qWZlLI/edit#heading=h.id8agdzq4ub

File: modules/angular2/src/core/di/binding.ts [hunk 5]
```diff
@@ -265,40 +286,12 @@ export class ResolvedFactory {
 }
 
 /**
- * Provides an API for imperatively constructing {@link Binding}s.
+ * Creates a {@link Binding}.
  *
- * To construct a {@link Binding}, bind a `token` to either a class, a value or a factory function.
+ * To construct a {@link Binding}, bind a `token` to either a class, a value, or a factory function.
```
Comment (jeffbcross) [inline modules/angular2/src/core/di/binding.ts]: Or to an alias to another token?

File: modules/angular2/src/core/di/exceptions.ts [hunk 6]
```diff
@@ -32,9 +32,22 @@ function constructResolvingPath(keys: any[]): string {
  * Base class for all errors arising from misconfigured bindings.
  */
 export class AbstractBindingError extends BaseException {
+<<<<<<< HEAD
```
Comment (jeffbcross) [inline modules/angular2/src/core/di/exceptions.ts]: Fix!

File: modules/angular2/src/core/di/forward_ref.ts [hunk 7]
```diff
@@ -5,25 +5,27 @@ export interface ForwardRefFn { (): any; }
 /**
  * Allows to refer to references which are not yet defined.
  *
- * This situation arises when the key which we need to refer to for the purposes of DI is declared,
+ * For instance, `forwardRef` is used when the key which we need to refer to for the purposes of DI is declared,
```
Comment (jeffbcross) [inline modules/angular2/src/core/di/forward_ref.ts]: Could you add a little bit more about common cases when this occurs?

**Comment by jeffbcross** [comment]
LGTM with comments addressed.

**Comment by vsavkin** [comment]
Merged

**Pull request closed without merging**